### PR TITLE
Feature/s4 add listing item

### PR DESCRIPTION
**Pull request by LHuss**
For issue #48, 

- refactored Listing to match instance variables

- adding an item to the database when pressing the add item button

- generating UUID for the item's ID. This will avoid any key repetition in the foreseeable future ("In other words, only after generating 1 billion UUIDs every second for the next 100 years, the probability of creating just one duplicate would be about 50%" -https://stackoverflow.com/questions/1155008/how-unique-is-uuid)

### CODE REVIEW

File: app/src/main/java/Fragments/AddItemFragment.java [hunk 1]
```diff
@@ -85,89 +87,94 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
-//
-//                }
-//
-//            }
-//        });
+        rootRef = FirebaseDatabase.getInstance().getReference();
+        databaseItems = rootRef.child("Items");
+
+        mStorage = FirebaseStorage.getInstance().getReference();
+
```
Comment (kkhh001) [inline app/src/main/java/Fragments/AddItemFragment.java:94]: Good use of the naming conventions 👍

File: app/src/main/java/Fragments/AddItemFragment.java [hunk 2]
```diff
@@ -85,89 +87,94 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+                    ItemDescription listingItem = new ItemDescription(id, itemName, itemPrice, "", itemDescription);
+
+                    databaseItems.child(id).child("Price").setValue(itemPrice);
+                    databaseItems.child(id).child("Description").setValue(itemDescription);
+                    databaseItems.child(id).child("Name").setValue(itemName);
+                    databaseItems.child(id).child("ImageURL").setValue("");
+                    databaseItems.child(id).child("OwnerID").setValue("userUID");
+
+                    Toast.makeText(getContext(), "Item Posted", Toast.LENGTH_LONG).show();
+
```
Comment (kkhh001) [inline app/src/main/java/Fragments/AddItemFragment.java:173]: Would we be able to redirect the user to the home listings page once the adding item function is completed? Though I do like the notification which tells the function of adding is complete!

**Review by kkhh001 — APPROVED** [review]
Really nicely implemented 👍 I like the fact that you can add an image of the item from two different apps. Good job!

**Review by Alex-Hayeur — COMMENTED** [review]
I have a similar question to Kayne about where the user ends up after adding an item, but otherwise good code. The many changes to follow the naming convention are appreciated.

**Review by Alex-Hayeur — APPROVED** [review]

File: app/src/main/java/Fragments/AddItemFragment.java (hunk 2 above)
Comment (kkhh001) [reply app/src/main/java/Fragments/AddItemFragment.java:173]: So worst case we can make the tweak (redirecting to listing page after adding an item) later!

**Pull request merged**